### PR TITLE
Add/improve doc comments & format a bit

### PR DESCRIPTION
**Pull request by konsumlamm**

### CODE REVIEW

File: src/bigints.nim [hunk 1]
```diff
@@ -13,12 +12,12 @@ type
 proc normalize(a: var BigInt) =
   for i in countdown(a.limbs.high, 0):
     if a.limbs[i] > 0'u32:
-      a.limbs.setLen(i+1)
+      a.limbs.setLen(i + 1)
```
Comment (xflywind) [inline src/bigints.nim:15]: > Use a..b instead of a .. b, except when b contains an operator, for example a .. -3. Likewise with a..<b, a..^b and other operators starting with ...

ref https://nim-lang.org/docs/nep1.html#introduction-miscellaneous

better to leave it uncahnged

File: src/bigints.nim [hunk 2]
```diff
@@ -13,12 +12,12 @@ type
...
       return
   a.limbs.setLen(1)
 
 proc initBigInt*(vals: sink seq[uint32], isNegative = false): BigInt =
-  ## Initialize BigInt from a sequence of `uint32` values.
+  ## Initialize a `BigInt` from a sequence of `uint32` values.
```
Comment (xflywind) [inline src/bigints.nim:20]: pre-existing issue
```suggestion
  ## Initializes a `BigInt` from a sequence of `uint32` values.
```

File: src/bigints.nim [hunk 3]
```diff
@@ -477,7 +479,7 @@ proc shiftRight(a: var BigInt, b: BigInt, c: int) =
     a.limbs.setLen(a.limbs.high)
 
 proc `shr`*(x: BigInt, y: int): BigInt =
-  ## Computes a right shift of a `BigInt`.
+  ## Shift a `BigInt` to the right (arithmetically).
```
Comment (xflywind) [inline src/bigints.nim:482]:
```suggestion
  ## Shifts a `BigInt` to the right (arithmetically).
```

File: src/bigints.nim [hunk 4]
```diff
@@ -498,7 +501,7 @@ proc pow*(x: BigInt, y: int): BigInt =
     base *= tmp
 
 proc `shl`*(x: BigInt, y: int): BigInt =
-  ## Computes a left shift of a `BigInt`.
+  ## Shift a `BigInt` to the left.
```
Comment (xflywind) [inline src/bigints.nim:504]:
```suggestion
  ## Shifts a `BigInt` to the left.
```

File: src/bigints.nim [hunk 5]
```diff
@@ -605,7 +611,7 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
     var a = zero
     a.limbs.setLen(k)
     let wm1 = r.limbs[r.limbs.high]
-    let wm2 = r.limbs[r.limbs.high-1]
+    let wm2 = r.limbs[r.limbs.high - 1]
```
Comment (xflywind) [inline src/bigints.nim:614]: ditto

File: src/bigints.nim (hunk 1 above)
Comment (konsumlamm) [reply src/bigints.nim:15]: That document only talks about operators starting with `..`. It was my impression that for other operators, spaces around it are generally preferred.

File: src/bigints.nim [hunk 6]
```diff
@@ -1,8 +1,7 @@
-## The official library for arbitrary precision integers, implemented in pure
-## Nim, without any external dependencies.
+## Arbitrary precision integers.
```
Comment (narimiran) [inline src/bigints.nim:1]: Why this change? What's wrong with the existing description?

File: src/bigints.nim (hunk 1 above)
Comment (narimiran) [reply src/bigints.nim:15]: I'm usually quite pedantic when it comes to whitespace, trying to make code "pretty".

In most of these cases (especially inside of square brackets, i.e. sequence index) where there is no whitespace, it was done on purpose.

Comment (narimiran) [reply src/bigints.nim:15]:
```suggestion
      a.limbs.setLen(i+1)
```

File: src/bigints.nim (hunk 5 above)
Comment (narimiran) [reply src/bigints.nim:614]:
```suggestion
    let wm2 = r.limbs[r.limbs.high-1]
```

File: src/bigints.nim [hunk 7]
```diff
@@ -617,11 +623,11 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
       # estimate quotient digit, may rarely overestimate by 1
       let vtop = q.limbs[v + dn]
       assert vtop <= wm1
-      let vv = (uint64(vtop) shl 32) or q.limbs[v+dn-1]
+      let vv = (uint64(vtop) shl 32) or q.limbs[v + dn - 1]
```
Comment (narimiran) [inline src/bigints.nim:626]:
```suggestion
      let vv = (uint64(vtop) shl 32) or q.limbs[v+dn-1]
```

File: src/bigints.nim [hunk 8]
```diff
@@ -617,11 +623,11 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
...
       var q1 = uint64(vv) div wm1
       var r1 = uint64(vv) mod wm1
 
-      while (uint64(wm2)*q1) > ((r1 shl 32) or q.limbs[v+dn-2]):
+      while (uint64(wm2) * q1) > ((r1 shl 32) or q.limbs[v + dn - 2]):
```
Comment (narimiran) [inline src/bigints.nim:630]:
```suggestion
      while (uint64(wm2) * q1) > ((r1 shl 32) or q.limbs[v+dn-2]):
```

File: src/bigints.nim [hunk 9]
```diff
@@ -644,9 +650,9 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
         z += qib
 
         if z < 0:
-          q.limbs[v+i] = not z.limbs[0] + 1
+          q.limbs[v + i] = not z.limbs[0] + 1
```
Comment (narimiran) [inline src/bigints.nim:653]:
```suggestion
          q.limbs[v+i] = not z.limbs[0] + 1
```

File: src/bigints.nim [hunk 10]
```diff
@@ -644,9 +650,9 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
...
         else:
-          q.limbs[v+i] = z.limbs[0]
+          q.limbs[v + i] = z.limbs[0]
```
Comment (narimiran) [inline src/bigints.nim:655]:
```suggestion
          q.limbs[v+i] = z.limbs[0]
```

File: src/bigints.nim [hunk 11]
```diff
@@ -663,7 +669,7 @@ proc unsignedDivRem(q, r: var BigInt, n, d: BigInt) =
       if vtop.initBigInt + zhi < 0:
         carry = 0
         for i in 0 ..< dn:
-          carry += q.limbs[v+i]
+          carry += q.limbs[v + i]
```
Comment (narimiran) [inline src/bigints.nim:672]:
```suggestion
          carry += q.limbs[v+i]
```

File: src/bigints.nim (hunk 6 above)
Comment (konsumlamm) [reply src/bigints.nim:1]: IMO this belongs in the package description, if anywhere, not the module description (especially that it's official), since it's about the package and not really the module.

Comment (konsumlamm) [reply src/bigints.nim:1]: @narimiran so, what do you think?
